examples/composition.rs

```diff
@@ -0,0 +1,593 @@
+        //             queue,
+        //             hwnd,
+        //             &SwapchainDesc {
+        //                 width: 1024,
+        //                 height: 1024,
+        //                 format: DXGI_FORMAT_B8G8R8A8_UNORM,
+        //                 stereo: false,
+        //                 sample: SampleDesc {
+        //                     count: 1,
+        //                     quality: 0,
+        //                 },
+        //                 buffer_usage: DXGI_USAGE_RENDER_TARGET_OUTPUT,
+        //                 buffer_count: NUM_OF_FRAMES as _,
+        //                 scaling: Scaling::Stretch,
+        //                 swap_effect: SwapEffect::FlipSequential,
+        //                 alpha_mode: AlphaMode::Ignore,
+        //                 flags: 0,
+        //             },
+        //         );
+        //         (hr == 0).then(|| ptr)
+        //     }
+        //     .expect("Unable to create swapchain");
+        //     let (ptr, hr) = unsafe { sw.cast::<IDXGISwapChain3>() };
+        //     (hr == 0).then(|| ptr)
+        // }
+        // .expect("Unable to cast swapchain");
+
+        // Create heap descriptor
+        let desc_heap = {
+            let (ptr, hr) = device.create_descriptor_heap(
+                NUM_OF_FRAMES as _,
+                DescriptorHeapType::Rtv,
+                DescriptorHeapFlags::empty(),
+                0, /* node mask */
+            );
+            (hr == 0).then(|| ptr)
+        }
+        .expect("Unable to create heap descriptor thing");
+
+        // Create resource per frame
+        let mut descriptor: CpuDescriptor = desc_heap.start_cpu_descriptor();
+        let _descriptor_inc_size = device.get_descriptor_increment_size(DescriptorHeapType::Rtv);
+        println!("what {}", _descriptor_inc_size);
+        let resources = (0..NUM_OF_FRAMES)
+            .map(|i| {
+                let resource = {
+                    let (ptr, hr) = swap_chain.as_swapchain0().get_buffer(i as _);
+                    (hr == 0).then(|| ptr)
+                }
+                .expect("Unable to create resource");
+
+                unsafe {
+                    device.CreateRenderTargetView(resource.as_mut_ptr(), 0 as _, descriptor);
+                    descriptor.ptr += _descriptor_inc_size as usize;
+                }
+
+                resource
+            })
+            .collect::<Vec<_>>()
+            .try_into()
+            .expect("Unable to get resources as array");
+
+        // self.factory = Some(factory);
+        self.adapter = Some(adapter);
+        self.device = Some(device);
+        self.queue = Some(queue);
+        self.allocator = Some(allocator);
+        self.desc_heap = Some(desc_heap);
+        self.desc_size = Some(_descriptor_inc_size);
+        self.swap_chain = Some(swap_chain);
+        self.comp_device = Some(comp_device);
+        self.resources = Some(resources);
+    }
+
+    pub fn load_assets(&mut self) {
```

Comment:
this isn't required to demonstrate the composition. Just clearing the screen (or doing multiple clears of sub-rectangles) would do the trick, and would end up with less code here.

Reply:
Thanks for looking into this! 

I have no idea what I can remove code from this yet. Or how to "clear sub-rectangles"? Code in this example I basically copy & pasted blindly from gfx and microsoft's DirectX examples until I got something visible on screen. And I still have no idea what I can just "take away" and have working example.